File: mc.f90
program mc
    implicit none

    integer :: number_of_particles, number_of_steps, current_step
    real :: material_thickness, travel_distance, particle_position
    real :: theta, pi
    integer :: escaped_particles, i
    character(len=256) :: str

    pi = atan(1.0)*4.0

    open(1, file="input.txt")
    read(1,*) str, number_of_particles
    read(1,*) str, number_of_steps
    read(1,*) str, material_thickness
    read(1,*) str, travel_distance
    close(1)

    escaped_particles = 0

    call RANDOM_SEED()

    do i = 1, number_of_particles
        particle_position = 0.0
        current_step = 0

        do while (current_step < number_of_steps)
            call RANDOM_NUMBER(theta)
            theta = theta * 2.0 * pi

            particle_position = particle_position + travel_distance*cos(theta)

            if (particle_position < 0.0 .or. particle_position >= material_thickness) then
                current_step = number_of_steps
            else
                current_step = current_step + 1
            end if

        end do

        if (particle_position >= material_thickness) then
            escaped_particles = escaped_particles + 1
        end if

    end do

    print*, "Escaped Particles: ", escaped_particles
    print*, "Percent Escaped: ", 100.0*real(escaped_particles)/real(number_of_particles)

end program mc

File: mc.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <cmath>

using namespace std;

#define PI 3.14159

int main() {

    int number_of_particles;
    int number_of_steps;
    float material_thickness;
    float travel_distance;
    float particle_position;
    int current_step;
    float theta;
    int escaped_particles;

    string str;

    ifstream indata;
    indata.open("input.txt");
    indata >> str >> number_of_particles;
    indata >> str >> number_of_steps;
    indata >> str >> material_thickness;
    indata >> str >> travel_distance;
    indata.close();

    escaped_particles = 0;

    for (int i = 0; i < number_of_particles; i++) {
        particle_position = 0.0;
        current_step = 0;

        while (current_step < number_of_steps) {
            theta = ((float) rand()/RAND_MAX)*2*PI;

            particle_position += travel_distance*cos(theta);

            if (particle_position < 0 || particle_position > material_thickness) {
                current_step = number_of_steps;
            } else {
                current_step++;
            }
        }

        if (particle_position >= material_thickess) {
            escaped_particles += 1;
        }
    }

    cout << "Escaped Particles: " << escaped_particles << endl;
    cout << "Percent Escaped: " << double(escaped_particles)*100/double(number_of_particles)<<endl;
    return 0;
}